cpp/CMakeLists.txt

```diff
@@ -156,9 +160,28 @@ setup_cuda_architectures()
 find_package(CUDAToolkit 11.2 REQUIRED COMPONENTS cudart_static cuda_driver
```

Comment:
Can you make this `find_package` find the actual wanted components? Please also add the missing `nvrtc`/`nvrtc_static` as well. Thanks.

Reply:
If CUDA is installed via some package manager, both the static and shared version of the components may not be present. Please build a list consists of components that will be used.

Reply:
ok done